lib/beaker/dsl/roles.rb

```diff
@@ -115,7 +115,11 @@ def not_controller(host)
       #       puts "this host is ONLY an agent!"
       #     end
       def agent_only(host)
-        host['roles'].length == 1 && host['roles'].include?('agent')
+        if host[:pe_ver] && version_is_less(host[:pe_ver], '4.0')
+          false
```

Comment:
Yeah, I don't like mixing this logic into agent_only, which is only for indicating if a host has the single role 'agent'.

A separate method is in order for what you are trying to determine.